x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_config_panel.tsx

```diff
@@ -94,67 +84,161 @@ function newLayerState(layerId: string): LayerConfig {
 
+  const [localState, setLocalState] = useState({
+    isChartOptionsOpen: false,
+    openLayerId: null,
+  } as LocalState);
+
   return (
     <EuiForm className="lnsConfigPanel">
+      <EuiFormRow>
+        <EuiPopover
+          id="lnsXY_chartConfig"
+          isOpen={localState.isChartOptionsOpen}
+          closePopover={() => {
+            setLocalState({ ...localState, isChartOptionsOpen: false });
+          }}
+          button={
+            <EuiFormRow>
+              <>
+                <EuiButton
+                  iconType="gear"
+                  size="s"
+                  data-test-subj="lnsXY_chart_settings"
+                  onClick={() => {
+                    setLocalState({ ...localState, isChartOptionsOpen: true });
+                  }}
+                >
+                  <FormattedMessage
+                    id="xpack.lens.xyChart.chartSettings"
+                    defaultMessage="Chart Settings"
+                  />
+                </EuiButton>
+              </>
+            </EuiFormRow>
+          }
+        >
+          <>
+            <EuiFormRow>
+              <EuiSwitch
+                label={i18n.translate('xpack.lens.xyChart.isHorizontalSwitch', {
+                  defaultMessage: 'Rotate chart 90º',
+                })}
+                checked={state.isHorizontal}
+                onChange={() => {
+                  setState({
+                    ...state,
+                    isHorizontal: !state.isHorizontal,
+                  });
+                }}
+                data-test-subj="lnsXY_chart_horizontal"
+              />
+            </EuiFormRow>
+          </>
+        </EuiPopover>
+      </EuiFormRow>
+
       {state.layers.map((layer, index) => (
-        <EuiFormRow key={layer.layerId}>
+        <EuiFormRow key={layer.layerId} data-test-subj={`lnsXY_layer_${layer.layerId}`}>
           <EuiPanel>
-            <EuiFormRow
-              label={i18n.translate('xpack.lens.xyChart.layerLabel', {
-                defaultMessage: 'Layer',
-              })}
+            <EuiPopover
+              id="lnsXY_layer"
+              isOpen={localState.openLayerId === layer.layerId}
+              closePopover={() => {
+                setLocalState({ ...localState, openLayerId: null });
+              }}
+              className="eui-fullWidth"
+              anchorClassName="eui-fullWidth"
+              button={
+                <EuiFormRow>
+                  <EuiFlexGroup>
+                    <EuiFlexItem>
+                      <EuiButtonIcon
+                        iconType="gear"
+                        data-test-subj="lnsXY_layer_advanced"
+                        aria-label={i18n.translate('xpack.lens.xyChart.layerSettings', {
+                          defaultMessage: 'Edit layer settings',
+                        })}
+                        onClick={() => {
+                          setLocalState({ ...localState, openLayerId: layer.layerId });
```

Comment:
The one thing that's confusing about this is that only the gear icon is clickable, yet the popover is aligned to the whole panel header. I would make the whole header a clickable button.

<img src="https://d.pr/free/i/0P6H9t+" width="50%" />

Reply:
Talked about this separately, I will keep just the gear clickable for now and change the wrapping of what the popover anchor is.